app/v1/lifecycle/systems/endpoints.py

```diff
@@ -0,0 +1,40 @@
+from fastapi import APIRouter, Path
+
+from app.data.systems import OS_DATA_MOCKED
+
+v1_router = APIRouter()
+
+
+@v1_router.get("/systems")
+async def get_systems():
+    systems = get_systems_data()
+
+    return sorted(systems, key=lambda d: (d["major"], d["minor"]))
+
+
+@v1_router.get("/systems/{major}")
+async def get_systems_major(major: int = Path(..., description="Major version number")):
+    systems = get_systems_data(major)
+
+    return sorted(systems, key=lambda d: (d["major"], d["minor"]))
+
+
+@v1_router.get("/systems/{major}/{minor}")
+async def get_systems_major_minor(
+    major: int = Path(..., description="Major version number"),
+    minor: int = Path(..., description="Minor version number"),
```

Comment:
A [better way to handle](https://fastapi.tiangolo.com/tutorial/path-params-numeric-validations/?h=annotated#better-with-annotated) this is to [use Annotated](https://fastapi.tiangolo.com/python-types/#type-hints-with-metadata-annotations).
```suggestion
    major: t.Annotated[int, Path(description="Major version number")],
    minor: t.Annotated[int, Path(description="Minor version number")],
```

It's also unnecessary to pass the ellipsis to `Path`.